__tests__/UserCaption-test.js

```diff
@@ -1,12 +1,186 @@
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {username: 234, caption: 'corona is wack'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+  it('Throws failed propType Error for providing username value of number and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {username: 234.5, caption: 'corona is wack'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+  it('Throws failed propType Error for providing caption value of true and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {caption: true, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `caption` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+  it('Throws failed propType Error for providing caption value of false and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {caption: false, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `caption` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+  it('Throws failed propType Error for providing caption value of undefined and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {caption: undefined, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: The prop `caption` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
+    );
+  });
+  it('Throws failed propType Error for providing caption value of number and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {caption: 234, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `caption` of type `number` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+  it('Throws failed propType Error for providing caption value of number and not string', () => {
+    let result = checkPropTypes(
+      UserCaption.propTypes,
+      {caption: 234.5, username: 'emanasia'},
+      'prop',
+      UserCaption.username,
+    );
+    assert(
+      result ===
+        'Failed prop type: Invalid prop `caption` of type `number` supplied to `<<anonymous>>`, expected `string`.',
+    );
+  });
+});
 
```

Comment:
Can we combine all of these into a function, ie testPropTypes (username, caption, errormessage etc)
Since those are the default values that change with each test case?

Reply:
We can do this in a refactor process if needed ^

Reply:
Yes, great suggestion